BUILDER_DEV.md

```diff
@@ -1,390 +1,294 @@
+```
+$ sudo apt-get update
+```
+
+* Install git
+```
+$ sudo apt-get install git
+```
+
+* Install curl
+```
+$ sudo apt-get install curl
+```
+
+* Clone the [Habitat repo](https://github.com/habitat-sh/habitat)
+```
+$ git clone https://github.com/habitat-sh/habitat.git
+```
+
+* Install the [Habitat CLI](https://www.habitat.sh/docs/get-habitat/)
+* Copy habitat-builder-dev.2017-10-02.private-key.pem from 1Password into path/to/habitat/repo/.secrets/builder-github-app.pem (NOTE - this is required as of 12/11/17 - we are working on a solution to make access to the Habitat 1Password vault not required for dev environment setup).
+* Set up some environmental variables (I put these in my .bashrc)
+
+```
+export HAB_AUTH_TOKEN="<your github token>"
+export HAB_BLDR_URL="http://localhost:9636"
+export HAB_ORIGIN="core"
+```
+
+* symlink /src to path/to/your/habitat/repo
+
+```
+$ sudo ln -s path/to/your/habitat/repo /src
 ```
 
-`config_jobsrv.toml`
-```toml
-key_dir = "/path/to/bldr-key-pair"
+* Cd into your Habitat repo
 
-[archive]
-backend = "local"
-local_dir = "/tmp"
 ```
+$ cd path/to/your/habitat/repo
+```
+
+* Run the provision script
+
+```
+$ sudo ./support/linux/provision.sh
+```
+
+* After the script completes, source cargo/rust
+```
+$ source $HOME/.cargo/env
+```
+
+* Run the build-srv makefile (NOTE - if you receive an error about cargo not being found, make sure you have removed the "secure_path" section of /etc/sudoers)
 
-`config_sessionsrv.toml`
-```toml
-[permissions]
-admin_team = 1995301
-build_worker_teams = [2555389]
-early_access_teams = [1995301]
 ```
+$ sudo make build-srv
+```
+
+* Open up the builder-worker file
 
-`config_worker.toml`
-```toml
+```
+$ sudo vim /hab/svc/builder-worker
+```
+
+* Add your Github Auth Key:
+
+**/hab/svc/builder-worker/config.toml**
+```
+auth_token = ""
+```
+
+Should become:
+
+**/hab/svc/builder-worker/config.toml**
+```
 auth_token = "<your github token>"
-bldr_url = "http://localhost:9636"
-auto_publish = true
-airlock_enabled = false
 ```
 
+* Run this command (it may take awhile) to start all the Builder services
+
+```
+$ sudo -E make bldr-run
+```
+
+### UI Setup
+
+* Open a separate shell on your VM
+* Install npm
+
+```
+$ sudo apt-get install npm
+```
+
+* Cd into the builder-web component
+
+```
+$ cd path/to/your/habitat/repo/components/builder-web
+```
+
+* Install all dependencies
+
+```
+$ npm install
+```
+
+* Create the habitat.conf.js file
+
+```
+$ cp habitat.conf.sample.js habitat.conf.js
+```
+
+* Start the UI
+
+```
+$ sudo npm start
+```
+
+* Open up a browser and navigate to http://localhost:3000/#/pkgs - you should see the Habitat UI running locally.
+
+### Creating an Origin
 
-(Note: If you want your log files to persist across restarts of your development machine, replace `/tmp` with some other directory. It *must* exist and be writable before you start the job server).
+* Navigate to http://localhost:3000/#/pkgs
+* Log in
+* Click on "Create New Origin"
```

Comment:
Should this be "Navigate to http://localhost:3000/#/origins" and "Create origin"? That's what my UI looks like, but maybe I'm doing something different?